lib/Sema/CSSolver.cpp

```diff
@@ -1000,7 +1002,7 @@ static PotentialBindings getPotentialBindings(ConstraintSystem &cs,
     }
 
     if (exactTypes.insert(type->getCanonicalType()).second)
-      addPotentialBinding({type, kind, None});
+      addPotentialBinding({type, kind, None}, /*allowJoinMeet=*/!adjustedIUO);
```

Comment:
This is equivalent to `!alternateType`. Either works for me (with the current code having perhaps slightly more clarity given the naming), but I thought I would mention it.